subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/JavaDevelopmentKit.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain;
+
+import org.gradle.api.Incubating;
+import org.gradle.api.file.FileCollection;
+
+import java.io.File;
+
+/**
+ * Information about a Java development kit.
+ *
+ * @since 6.1
+ */
+@Incubating
+public interface JavaDevelopmentKit {
+    /**
+     * Returns the java compiler executable for this JDK.
+     */
+    File getJavacExecutable();
```

Comment:
Should these all go straight to either a `RegularFile` or a `Provider<RegularFile>` so they use the new types vs just File and `flatMap` or `map` don't require you to use the File-taking setter methods on other providers?

Reply:
yes, they probably should be